internal/host/host.go

```diff
@@ -1275,3 +1278,43 @@ func (m *Manager) GetHostByKubeKey(key types.NamespacedName) (*common.Host, erro
 func (m *Manager) UnRegisterHost(ctx context.Context, hostID, infraEnvID string) error {
 	return common.DeleteHostFromDB(m.db, hostID, infraEnvID)
 }
+
+func (m *Manager) GetKnownHostApprovedCounts(clusterID strfmt.UUID) (registered, approved int, err error) {
+	var hostCounts []struct {
+		Count    int
+		Approved bool
+	}
+	err = m.db.Table("hosts").Select("count(*) as count, approved").Group("approved").
+		Where("cluster_id = ? and status = ? and deleted_at is null", clusterID.String(), models.HostStatusKnown).Scan(&hostCounts).Error
+	if err != nil {
+		m.log.WithError(err).Errorf("Failed to get counts")
+		return
+	}
+	for _, h := range hostCounts {
+		registered += h.Count
+		if h.Approved {
+			approved += h.Count
+		}
+	}
+	return
+}
+
+func (m *Manager) HostWithCollectedLogsExists(clusterId strfmt.UUID) (bool, error) {
+	var hostExists []int
+	query := `SELECT 1 as host_exists 
+                  WHERE EXISTS 
+					  (SELECT 1 
```

Comment:
This indentation seems a bit extreme :sweat_smile: 

Was your editor not happy with this or something?

Reply:
Looks different in goland

Reply:
Seems to look differently depending on the viewer.
Tried to push another version. Hope it looks better now. Probably it will not be optimized for all viewers.